include/circt/Dialect/ESI/cosim/svdpi.h

```diff
@@ -0,0 +1,380 @@
+/* -*- C -*-
+ *
+ * svdpi.h
+ *
+ * SystemVerilog Direct Programming Interface (DPI).
+ *
+ * This file contains the constant definitions, structure definitions,
+ * and routine declarations used by SystemVerilog DPI.
+ *
+ * This file is from the SystemVerilog IEEE 1800-2017 Annex I.
+ */
+
+#ifndef CIRCT_DIALECT_ESI_COSIM_SVDPI_H
+#define CIRCT_DIALECT_ESI_COSIM_SVDPI_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Define size-critical types on all OS platforms. */
+#if defined(_MSC_VER)
+typedef unsigned __int64 uint64_t;
+typedef unsigned __int32 uint32_t;
+typedef unsigned __int8 uint8_t;
+typedef signed __int64 int64_t;
+typedef signed __int32 int32_t;
+typedef signed __int8 int8_t;
+#elif defined(__MINGW32__)
+#include <stdint.h>
+#elif defined(__APPLE__)
+#include <stdint.h>
+#elif defined(__linux) || (defined(__APPLE__) && defined(__MACH__))
+#include <inttypes.h>
+#else
+#include <sys/types.h>
+#endif
+
+/* Use to import a symbol into dll */
+#ifndef DPI_DLLISPEC
+#if (defined(_MSC_VER) || defined(__MINGW32__) || defined(__CYGWIN__))
+#define DPI_DLLISPEC __declspec(dllimport)
+#else
+#define DPI_DLLISPEC
+#endif
+#endif
+
+/* Use to export a symbol from dll */
+#ifndef DPI_DLLESPEC
+#if (defined(_MSC_VER) || defined(__MINGW32__) || defined(__CYGWIN__))
+#define DPI_DLLESPEC __declspec(dllexport)
+#else
+#define DPI_DLLESPEC
+#endif
+#endif
+
+/* Use to mark a function as external */
+#ifndef DPI_EXTERN
+#define DPI_EXTERN
+#endif
```

Comment:
Is DPI_EXTERN for ancient C compilers?  Why do we need it?

Reply:
I don't know. This is a standard header.

Reply:
k